src/components/TextInput.tsx

```diff
@@ -105,23 +106,27 @@ export const TextInput = (props: Props) => {
           minLength={props.minLength}
           name={props.name}
           ref={inputRef}
-          onChange={(e: any) => {
-            setInternalValue(e.target.value)
-            props.onChange?.(e.target.value)
+          onChange={(event: any) => {
+            const newValue = event.target.value
+
+            setInternalValue(newValue)
+            props.onChange?.(newValue)
             if (!isFocus) {
               // Required for browser auto-fill fields to ensure the form gets the values
-              props.onBlur?.(e.target.value)
+              props.onBlur?.(newValue)
             }
             // if (!props.onlyOnBlur) {
             //   // @ts-ignore
             //   props.onChange?.(e.target.value)
             // }
           }}
           onBlur={() => {
-            const v = (internalValue || '').trim()
-            setInternalValue(v)
+            let newValue = (internalValue || '').trim()
+            newValue = props.parseValue?.(newValue) || newValue
+
+            setInternalValue(newValue)
```

Comment:
I'd have just one state for the component where u have all the states. Instead of
const [isFocus, setIsFocus] = useState(false)
const [internalValue, setInternalValue] = useState(props.value)
  
const [state, setState] = useState({isFocus: false, internalValue: props.value})

Reply:
Yeah, let's refactor that 👍